### DISCUSSION TITLE
Cannot Navigate - Mushr Melodic

### DISCUSSION THREAD

**Discussion by jmcnulty7070** [prl-mushr/mushr#113] [Software]
Hi, I have been trying for a year to get Mushr code working on my Mushr 1/10 car. I can map and localize, but the code doesn’t work for navigation. (Stated as buggy by repo).

I would like to use the code here to achieve some form of navigation.

I am exhausted. 

Is there a repo that works in full with navigation for Mushr hardware? 

I have tried the docker version for noetic as well. 

I would be very grateful for help.

**Comment by schmittlema** [comment]
Hi @jmcnulty7070, I'm sorry to hear that, let's see if we can help get you moving. To start, we no longer support melodic and recommend upgrading to noetic.

[Tutorial on Noetic install
](https://mushr.io/tutorials/noetic_quickstart/)[Tutorial on navigation](https://mushr.io/tutorials/noetic-navigation/)

I'm assuming you followed these two tutorials. For the navigation tutorial, could you specify what step went wrong and what the error or issue was?

**Comment by jmcnulty7070** [comment]
Hi Matt,

Thank you.
I had trouble with visualization on a VM with docker. (RVIZ)
Can I even use a VM for my PC?
What Jetson Jetpack is needed for Noetic. I think I installed the
wrong one in the past.

**Reply by schmittlema** [reply]
Visualization 
It will be easiest with foxglove per the tutorial. rviz proved to be tricky unless you have ubuntu. If you have a ubuntu VM and want to try rviz then you likely havae the ROS_MASTER_URI and ROS_IP not set correctly outside of the container. If you paste the output of the following commands I can recommend the right settings
inside docker:
`echo $ROS_MASTER_URI && echo $ROS_IP`
`ifconfig`


Jetpack
The latest jetpack I believe should work with noetic. If not[ this version should](https://developer.nvidia.com/embedded/l4t/r32_release_v7.1/jp_4.6.1_b110_sd_card/jeston_nano/jetson-nano-jp461-sd-card-image.zip)

**Comment by jmcnulty7070** [comment]
Thank you Matt. Appreciate it!Sent from my iPhoneOn Jul 5, 2023, at 3:37 PM, Matt Schmittle ***@***.***> wrote:﻿
Visualization
It will be easiest with foxglove per the tutorial. rviz proved to be tricky unless you have ubuntu. If you have a ubuntu VM and want to try rviz then you likely havae the ROS_MASTER_URI and ROS_IP not set correctly outside of the container. If you paste the output of the following commands I can recommend the right settings
inside docker:
echo $ROS_MASTER_URI && echo $ROS_IP
ifconfig
Jetpack
The latest jetpack I believe should work with noetic. If not this version should
